server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,137 +257,398 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
         }
 
-        @Override
-        public void onFailure(Throwable t)
+        if (executorService != null) {
+          executorService.shutdownNow();
+        }
+
+        LOG.debug("Stopped");
+      }
+      catch (Exception ex) {
+        LOG.makeAlert(ex, "Got Exception while stop()").emit();
+      }
+      finally {
+        //so that subsequent start() would happen, even if stop() failed with exception
+        lifecycleLock.exitStop();
+        lifecycleLock.reset();
+      }
+    }
+  }
+
+  private void initializeLookupsConfigWatcher()
+  {
+    //Note: this call is idempotent, so multiple start() would not cause any problems.
+    lookupMapConfigRef = configManager.watch(
+        LOOKUP_CONFIG_KEY,
+        new TypeReference<Map<String, Map<String, LookupExtractorFactoryMapContainer>>>()
         {
-          backgroundManagerExitedLatch.countDown();
-          if (backgroundManagerFuture.isCancelled()) {
-            LOG.info("Background lookup manager exited");
-            LOG.trace(t, "Background lookup manager exited with throwable");
-          } else {
-            LOG.makeAlert(t, "Background lookup manager exited with error!").emit();
+        },
+        null
+    );
+
+    // backward compatibility with 0.10.0
+    if (lookupMapConfigRef.get() == null) {
+      Map<String, Map<String, Map<String, Object>>> oldLookups = configManager.watch(
+          OLD_LOOKUP_CONFIG_KEY,
+          new TypeReference<Map<String, Map<String, Map<String, Object>>>>()
+          {
+          },
+          null
+      ).get();
+
+      if (oldLookups != null) {
+        Map<String, Map<String, LookupExtractorFactoryMapContainer>> converted = new HashMap<>();
+        for (Map.Entry<String, Map<String, Map<String, Object>>> tierEntry : oldLookups.entrySet()) {
+          Map<String, Map<String, Object>> oldTierLookups = tierEntry.getValue();
+          if (oldTierLookups != null && !oldTierLookups.isEmpty()) {
+            Map<String, LookupExtractorFactoryMapContainer> convertedTierLookups = new HashMap<>();
+            for (Map.Entry<String, Map<String, Object>> e : oldTierLookups.entrySet()) {
+              convertedTierLookups.put(e.getKey(), new LookupExtractorFactoryMapContainer(null, e.getValue()));
+            }
+            converted.put(tierEntry.getKey(), convertedTierLookups);
           }
         }
-      });
-      started = true;
-      LOG.debug("Started");
+        configManager.set(
+            LOOKUP_CONFIG_KEY,
+            converted,
+            new AuditInfo("autoConversion", "autoConversion", "127.0.0.1")
+        );
+      }
     }
   }
 
-  @LifecycleStop
-  public void stop()
+  private void lookupManagementLoop()
   {
-    synchronized (startStopSync) {
-      if (!started) {
-        LOG.warn("Not started, ignoring stop request");
-        return;
+    // Sanity check for if we are shutting down
+    if (Thread.currentThread().isInterrupted() || !lifecycleLock.awaitStarted(1, TimeUnit.MILLISECONDS)) {
```

Comment:
well this was there even before this PR. also, I'm not sure why it will get cleared automatically in this case when exec was told to shutdown.
we have bigger problems if interrupt is cleared by itself... check here is just in the beginning, say interrupt happened after completing this line, loop is gonna generate a bunch of http requests which should all terminate when exec is asked to shutdown or else they might wait for timeout or a response from the lookup node.